array/DNAm/analysis/EWAS/dmrcate.r

```diff
@@ -0,0 +1,123 @@
+#---------------------------------------------------------------------#
+##
+## Title: Regional analysis for case control within each cell type
+##
+## Purpose of script: identify regions of differential  DNA methylation for
+## schizophrenia vs controls for each cell type separately using DMRcate
+##
+##---------------------------------------------------------------------#
+
+#----------------------------------------------------------------------#
+# LOAD PACKAGES
+#----------------------------------------------------------------------#
+
+library(DMRcate)
+
+#----------------------------------------------------------------------#
+# DEFINE FUNCTIONS
+#----------------------------------------------------------------------#
+
+gg_color_hue <- function(n) {
+  hues = seq(15, 375, length = n + 1)
+  hcl(h = hues, l = 65, c = 100)[1:n]
+}
+
+#----------------------------------------------------------------------#
+# DEFINE PARAMETERS
+#----------------------------------------------------------------------#
+
+args<-commandArgs(trailingOnly = TRUE)
+dataDir <- args[1]
+cellType <- args[2]
+refPath <- args[3]
+
+normData<-file.path(dataDir, "3_normalised/normalised.rdata")
+cellCompData<-file.path(dataDir, "4_analysis", "EstimatedNeuralCellComposition.rdata")
+resPath<-file.path(dataDir, "4_analysis/EWAS")
+
+#----------------------------------------------------------------------#
+# LOAD AND PREPARE DATA
+#----------------------------------------------------------------------#
+
+setwd(dataDir)
+load(normData)
+load(cellCompData)
+
+print(paste0("running DMRcate on ", cellType, " cell type..."))
+## subset to cell type samples
+QCmetrics<-QCmetrics[which(QCmetrics$Cell.type == cellType),]
+
+# filter to schizophrenia and control only
+QCmetrics<-QCmetrics[QCmetrics$Phenotype %in% c("Schizophrenia", "Control"),]
```

Comment:
The hard coding of "schizophrenia" here prevents this pipeline form being used for other cell sorted data projects

Reply:
Yes - not ideal. For now I am ok with it as these scripts document the analysis for that project. If we were to generalise we would need a config file or something to document that analysis.